components/map.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Mapsearch from './mapsearch.jsx';
+
+class Map extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onSearch = this.onSearch.bind(this);
+  }
+
+  componentDidMount() {
+    const theMap = this.refs.map;
+    const mapOptions = {
+      center: {lat: 37.773972,
+      lng: -122.431297},
+      zoom: 12,
+      minZoom: 3
+    };
+
+    this.initializeMap();
```

Comment:
can we move this to constructor
we have our code setup so the map will mount and unmount quite often, so it will take a lot more computer work to continuously initialize the map. Move to constructor do it doesn't rerender every time